src/Swarm/Game/Step.hs

```diff
@@ -913,6 +954,10 @@ execConst c vs k = do
           [ "Bad application of execConst:"
           , from (prettyCEK (Out (VCApp c (reverse vs)) k))
           ]
+  finishCookingRecipe r = do
+    let recTime = pred $ r ^. recipeTime
```

Comment:
Well I am certainly not against it, as now I can just always return waiting. :slightly_smiling_face:

Reply:
I still think `pred` is wrong, it should just be `recTime = r ^. recipeTime`.  For example, suppose `r ^. recipeTime = 1`.  Then we want to do `Waiting (time + 1)`, not `Waiting (time + 0)`, since the first one means we should resume on the next tick, whereas the second one does not even make sense, it means we want to wait and continue running once the time reaches... now.